specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessments.json

```diff
@@ -0,0 +1,330 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Security Center",
+    "description": "API spec for Microsoft.Security (Azure Security Center) resource provider",
+    "version": "2019-01-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
```

Comment:
Delete API for deleting an assessment missing. If user can create it, he should have a way to delete it.

Reply:
Hi @ravbhatnagar. Delete scenarios occur only when the base resource is deleted (resource the assessment refers to). There is no way to manually delete an assessment, by design.

Reply:
Isnt that design flawed then? If the customer can explicitly create it, he should be able to remove what he created. The customer may not want to delete the resource the assessment refers to since it may be used for other purposes. 
Lets say the customer created a new assessment and does not want to see the older one anymore. I am assuming this assessment for a particular resource is not singleton.

Reply:
this is indeed for particular resource
if the user created a new assessment then he can remove the assessment type as a whole by deleting the relevant "assessmentMetadata"
it will result in auto deletion of all the linked assessments

Reply:
I understand your cascaded auto deletion scenario, but you haven't addressed Gaurav's point about a design flaw with regard to your scenario. 

As a side note: It doesn't look like your assessmentMetadata PR allows delete either.
https://github.com/Azure/azure-rest-api-specs/pull/7622/files

Reply:
added the delete method to the metadata API

sorry for not being clear, the assessment for a particular resource **is** a singleton (not as Gaurav assumed)
for example, a user cannot report a single VM as not encrypted more than once or with two different states
so if a user want to change the state or any other property of the assessment he can just override it
and if the user wants to delete the assessment type completely he can delete the relevant metadata